### ISSUE TITLE
Multi-line input only processes the first line

### ISSUE THREAD

**Issue by languitar** [ipython/ipython#9948]
With ipython 5.0 and 5.1 (from anaconda) I have the issue that multiple lines of input in one batch via the tty are not processed correctly. Only the first line is evaluated. This can be verified via tmux and the send-keys feature:

Start ipython in a tmux session:
```
tmux new -s test ipython
```

And send two lines of code in one batch from another terminal:
```
tmux send-keys -t 'test' 'print 42' Enter 'print 44' Enter
```

Only the first line is evaluated.

**Comment by Carreau** [comment]
Hum, that's probably because the event arrive to fast for processing. 
You are basically typing `print 44` while IPython in in between two prompt, as "enter" is both new line and execute (depending if the input is "complete" or not).

if you try `tmux send-keys -t 'test' 'for i in range(10):' Enter 'print 44' Enter` if should work.  

We need to bind a shortcut that adds a new line without executing, if you are scripting you can use `C-o Down` to force a newline and `Escape Enter` to force execution.

**Comment by languitar** [comment]
My expectation would be that the unprocessed characters after the first enter are just queued up and processed once the first execution returns. This is the behavior that existed in previous versions and which I am used to from basically any other command-line based tool. Simply swallowing key presses doesn't sound nice to me.

**Comment by Carreau** [comment]
I agree, I was just offering a work around. Though because of IPython
architecture it might not be that easy to fix.

The --simple-prompt option might fix that as well, but then you will lose
most of the bells and whistles, syntactic coloration and multiline edit.

Unfortunately there is a trade-off

**Comment by ivanov** [comment]
This issue may be the same as the one described in #9948. I want to address this by tackling #10364.

**Comment by jonathanslenders** [comment]
I guess the problem here is the ``send-keys`` command of tmux doesn't send keystrokes using bracketed paste. If you wrap everything in bracketed paste characters, then I think it should work.

**Comment by groner** [comment]
I think paste is just one way of demonstrating the problem.

Using `ipython -im MODULE` (or `ipython -i SCRIPT`) the repl can be delayed while the module or script is loaded, making it easy to type in multiple lines without using paste at all.
